cluster/plan.go

```diff
@@ -509,8 +509,8 @@ func (c *Cluster) BuildKubeletProcess(host *hosts.Host, serviceOptions v3.Kubern
 		if err != nil {
 			logrus.Debugf("Error while parsing cluster version: %s", err)
 		}
-		// cri-dockerd must be enabled if the cluster version is 1.24 and higher
-		if parsedRangeAtLeast124(parsedVersion) {
+
+		if useNewCRIEndpoint(c.Version, parsedVersion) {
```

Comment:
the variable `parsedVersion` is used only in this function; we can move lines 508-511 into the function `useNewCRIEndpoint`
```suggestion
		if useNewCRIEndpoint(c.Version) {
```